src/common/conditional_compilation/include/openvino/cc/factory.h

```diff
@@ -31,11 +31,16 @@ class Factory<Key, T(Args...)> {
 #ifdef SELECTIVE_BUILD
 #    define registerNodeIfRequired(Module, Name, key, Impl) \
         OV_PP_EXPAND(OV_PP_CAT(registerImpl, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(Module, _, Name))) < Impl > (key))
-#    define createNodeIfRegistered(Module, key, ...) createImpl(key, __VA_ARGS__)
+#    define registerNodeWithImplIfRequired(Module, Name, key, Impl) \
+        OV_PP_EXPAND(OV_PP_CAT(registerImpl, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(Module, _, Name)))(key, Impl))
+#    define createNodeIfRegistered(Module, key, ...)  createImpl(key, __VA_ARGS__)
```

Comment:
Can we define this macro as createNodeIfRegistered(Module, ...) createImpl(__VA_ARGS__)? So the createNodeNoArgsIfRegistered is redundant.

Reply:
Good suggestion, will be done!